src/main/java/io/vertx/ext/web/handler/impl/StaticHandlerImpl.java

```diff
@@ -260,18 +262,75 @@ private void resetTuning() {
     numServesBlocking = 0;
   }
 
+  private static final Pattern RANGE = Pattern.compile("^bytes=(\\d+)-(\\d*)$");
+
   private void sendFile(RoutingContext context, String file, FileProps fileProps) {
     HttpServerRequest request = context.request();
+
+    // check if the client is making a range request
+    String range = request.getHeader("Range");
+    // parse it
+    Long offset = null;
+    // end byte is length - 1
+    Long end = fileProps.size() - 1;
+
+    if (range != null) {
+      Matcher m = RANGE.matcher(range);
+      if (m.matches()) {
+        try {
+          String part = m.group(1);
+          // offset cannot be empty
+          offset = Long.parseLong(part);
+          // offset must fall inside the limits of the file
+          if (offset < 0 || offset >= fileProps.size()) {
+            throw new IndexOutOfBoundsException();
+          }
+          // length can be empty
+          part = m.group(2);
+          if (part != null && part.length() > 0) {
+            // ranges are inclusive
+            end = Long.parseLong(part);
+            // offset must fall inside the limits of the file
+            if (end < offset || end >= fileProps.size()) {
+              throw new IndexOutOfBoundsException();
+            }
+          }
+        } catch (NumberFormatException | IndexOutOfBoundsException e) {
+          context.fail(416);
+          return;
+        }
+      }
+    }
+
     writeCacheHeaders(request, fileProps);
 
+    // notify client we support range requests
+    MultiMap headers = request.response().headers();
+    headers.set("Accept-Ranges", "bytes");
+    // send the content length even for HEAD requests
+    headers.set("Content-Length", Long.toString(end + 1 - (offset == null ? 0 : offset)));
```

Comment:
Core will only set the header if the data is really sent, for a HEAD request since there is no data being sent the content-length is always 0, while is should be the length of the file.

I could move it to the if statement bellow, however core does check if the header is already set and only if not set will calculate it and set.